From 055d909aee76d36814723d81cfa98377657c2251 Mon Sep 17 00:00:00 2001
From: Matthias Runge <mrunge@redhat.com>
Date: Tue, 10 Dec 2013 10:37:55 +0100
Subject: [PATCH] Run selenium tests only when requested

In some environments, it's not wanted or possible
to execute selenium tests. These are only executed, if
--with-selenium is specified.

Change-Id: I39c40ac1feb64565f0bcbc97f6345d203b1d816e
Closes-Bug: 1259072
(cherry picked from commit f95b2b68d2219f717ae0e604af7e61e1a24dde6e)
---
 openstack_dashboard/dashboards/admin/projects/tests.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/openstack_dashboard/dashboards/admin/projects/tests.py b/openstack_dashboard/dashboards/admin/projects/tests.py
index 961af23..ba4803e 100644
--- a/openstack_dashboard/dashboards/admin/projects/tests.py
+++ b/openstack_dashboard/dashboards/admin/projects/tests.py
@@ -17,11 +17,13 @@
 import copy
 import datetime
 import logging
+import os
 
 from django.core.urlresolvers import reverse  # noqa
 from django import http
 from django.test.utils import override_settings  # noqa
 from django.utils import timezone
+from django.utils import unittest
 
 from mox import IgnoreArg  # noqa
 from mox import IsA  # noqa
@@ -35,7 +37,10 @@ from openstack_dashboard.test import helpers as test
 from openstack_dashboard import usage
 from openstack_dashboard.usage import quotas
 
-from selenium.webdriver import ActionChains  # noqa
+with_sel = os.environ.get('WITH_SELENIUM', False)
+if with_sel:
+    from selenium.webdriver import ActionChains  # noqa
+
 from socket import timeout as socket_timeout  # noqa
 
 
@@ -1509,6 +1514,8 @@ class UsageViewTests(test.BaseAdminViewTests):
         self.assertContains(res, '%s\r\n' % hdr)
 
 
+@unittest.skipUnless(os.environ.get('WITH_SELENIUM', False),
+                     "The WITH_SELENIUM env variable is not set.")
 class SeleniumTests(test.SeleniumAdminTestCase):
     @test.create_stubs(
         {api.keystone: ('tenant_list', 'tenant_get', 'tenant_update')})
